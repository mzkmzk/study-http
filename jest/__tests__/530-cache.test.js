const curl = require('../curl/index')

it('正常测试', async function(){
  const curlString = "curl -vs \
    -H 'Host: study-http.404mzk.com' \
    http://47.115.81.18:9500/test.log"
  const { status, body, row } = await curl(curlString)

  expect(status).toBe(200)
  expect(row).toMatchSnapshot()
})

it('if-none-match测试', async function(){
  const curlString = "curl -vs \
    -H 'If-None-Match: " + '"5ed8f9a5-f"' + "' \
    -H 'Host: study-http.404mzk.com' \
    http://47.115.81.18:9500/test.log"
  const { status, body, row } = await curl(curlString)

  expect(status).toBe(304)
  expect(row).toMatchSnapshot()
})