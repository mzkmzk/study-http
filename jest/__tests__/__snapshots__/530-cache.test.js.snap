// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if-none-match测试 1`] = `
"GET /test.log HTTP/1.1
User-Agent: curl/7.38.0
Accept: */*
If-None-Match: \\"5ed8f9a5-f\\"
Host: study-http.404mzk.com

HTTP/1.1 304 Not Modified
Server: nginx/1.17.10
Date: mock-time
Last-Modified: mock-time
Connection: keep-alive
ETag: \\"5ed8f9a5-f\\"

"
`;

exports[`正常测试 1`] = `
"GET /test.log HTTP/1.1
User-Agent: curl/7.38.0
Accept: */*
Host: study-http.404mzk.com

HTTP/1.1 200 OK
Server: nginx/1.17.10
Date: mock-time
Content-Type: application/octet-stream
Content-Length: 15
Last-Modified: mock-time
Connection: keep-alive
ETag: \\"5ed8f9a5-f\\"
Accept-Ranges: bytes

123456789ABCDEF"
`;
